checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -220,6 +225,78 @@ protected void constructorFromUsePreSubstitution(
     super.constructorFromUsePreSubstitution(tree, type);
   }
 
+  /**
+   * Class to implement the customized semantics of {@link MustCallAlias} (and {@link PolyMustCall})
+   * annotations; see the {@link MustCallAlias} documentation for details.
+   */
+  private class MustCallQualifierPolymorphism extends DefaultQualifierPolymorphism {
+    /**
+     * Creates a {@link MustCallQualifierPolymorphism}.
+     *
+     * @param env the processing environment
+     * @param factory the factory for the current checker
+     */
+    public MustCallQualifierPolymorphism(ProcessingEnvironment env, AnnotatedTypeFactory factory) {
+      super(env, factory);
+    }
+
+    @Override
+    protected void replace(
+        AnnotatedTypeMirror type, AnnotationMirrorMap<AnnotationMirror> replacements) {
+      boolean replacementsModified = false;
+      AnnotationMirror extantPolyAnnoReplacement = null;
+      TypeElement typeElement = TypesUtils.getTypeElement(type.getUnderlyingType());
+      // only customize replacement for type elements
+      if (typeElement != null) {
+        // TODO: should we assert that replacements.size() == 1?
```

Comment:
@smillst for the must-call type system we just have `@PolyMustCall` as a polymorphic annotation (with alias `@MustCallAlias`), with no parameters.  Given that, is it guaranteed that at this point, `replacements` will always have exactly one entry?

Reply:
Yes, it will have exactly one entry.